router/main.go

```diff
@@ -40,17 +40,19 @@ func main() {
 
 func makeHandler(c *http.Client, timeout time.Duration, upstreamURL string) func(w http.ResponseWriter, r *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
-
+		var host string
 		if len(r.Host) == 0 {
```

Comment:
Did this guard not work? Was `r.Host` actually `nil`?

Reply:
`r.Host[0:strings.Index(r.Host, ".")]`  panics when r.Host == "" maybe even the thing I provided is not guard enough for example what happens when we have string that does not have `.` I haven't really researched that case

Reply:
Sounds good. Do your tests cover both scenarios?